pkg/reconciler/instances/cleaner/action.go

```diff
@@ -16,17 +27,117 @@ func (a *CleanupAction) Run(context *service.ActionContext) error {
 		return nil
 	}
 
+	context.Logger.Infof("Action '%s' executed: passed version was '%s', passed type was %s", a.name, context.Task.Version, context.Task.Type)
+
 	if _, err := context.KubeClient.Clientset(); err != nil { //cleaner how to retrieve native Kubernetes GO client
 		return err
 	}
 
-	context.Logger.Infof("Action '%s' executed: passed version was '%s', passed type was %s", a.name, context.Task.Version, context.Task.Type)
-
 	namespaces := []string{"kyma-system", "kyma-integration"}
-	cliCleaner, err := cleanup.NewCliCleaner(context.Task.Kubeconfig, namespaces, context.Logger)
+
+	var kymaCRDsFinder cleanup.KymaCRDsFinder = func() ([]schema.GroupVersionResource, error) {
+		crdManifests, err := context.ChartProvider.RenderCRD(context.Task.Version)
+		if err != nil {
+			return nil, err
+		}
+		return findKymaCRDs(crdManifests, context.Logger)
+	}
+
+	dropFinalizersOnlyForKymaCRs := readDeleteStrategy(context.Task.Configuration) != "all"
```

Comment:
Can we maybe move the KEY + VALUE as public constant to it and use in the CLI to set this config-parameter in the task?

@clebs : JFYI